File: jambeth/jambeth-ioc/src/main/java/de/osthus/ambeth/converter/StringToIntArrayConverter.java
package de.osthus.ambeth.converter;

import de.osthus.ambeth.util.IDedicatedConverter;

public class StringToIntArrayConverter implements IDedicatedConverter
{
	@Override
	public Object convertValueToType(Class<?> expectedType, Class<?> sourceType, Object value, Object additionalInformation)
	{
		if (int[].class.equals(expectedType))
		{
			String[] split = StringToPatternConverter.splitPattern.split((String) value);
			int[] result = new int[split.length];
			for (int a = split.length; a-- > 0;)
			{
				result[a] = Integer.parseInt(split[a]);
			}
			return result;
		}
		StringBuilder sb = new StringBuilder();
		int[] array = (int[]) value;
		for (int item : array)
		{
			if (sb.length() > 0)
			{
				sb.append(StringToPatternConverter.splitPattern.pattern());
			}
			sb.append(item);
		}
		return sb.toString();
	}
}

File: ambeth/Ambeth.IoC/ambeth/converter/StringToIntArrayConverter.cs
using De.Osthus.Ambeth.Util;
using System;
using System.Text;

namespace De.Osthus.Ambeth.Converter
{
    public class StringToIntArrayConverter : IDedicatedConverter
    {
        public Object ConvertValueToType(Type expectedType, Type sourceType, Object value, Object additionalInformation)
        {
            if (typeof(int[]).Equals(expectedType))
            {
                String[] split = StringToPatternConverter.splitPattern.Split((String)value);
                int[] result = new int[split.Length];
                for (int a = split.Length; a-- > 0; )
                {
                    result[a] = Int32.Parse(split[a]);
                }
                return result;
            }
            StringBuilder sb = new StringBuilder();
            int[] array = (int[])value;
            foreach (int item in array)
            {
                if (sb.Length > 0)
                {
                    sb.Append(StringToPatternConverter.splitPattern.ToString());
                }
                sb.Append(item);
            }
            return sb.ToString();
        }
    }
}
